app/decorators/item_decorator.rb

```diff
@@ -0,0 +1,32 @@
+class ItemDecorator < Draper::Decorator
+  delegate_all
+  include Draper::LazyHelpers
+
+  def edit_link
+    link_to 'Edit', edit_item_path(item) if is_admin?
+  end
+
+  def delete_link
+    if is_admin?
+      link_to 'Destroy', item, method: :delete, data: { confirm: 'Are you sure?' }
+    end
+  end
+
+  def run_draw_or_bid
+    if is_admin?
+      link_to 'Run draw', draw_item_path(item), method: :post if can_run_draw?
+    else
```

Comment:
you can also write it here like:

```ruby
if can_run_draw? 
  link_to 'Run draw', draw_item_path(item), method: :post if can_run_draw?
elsif current_user
   link_to 'Bid It', bids_path(bid: { item_id: item.id }), method: :post
end
```

and in method `can_run_draw?` check if user is admin